pkg/storage/pebble_mvcc_scanner.go

```diff
@@ -916,7 +917,11 @@ func (p *pebbleMVCCScanner) getOne(ctx context.Context) (ok, added bool) {
 	}
 	if len(p.meta.RawBytes) != 0 {
 		// 7. Emit immediately if the value is inline.
-		return p.add(ctx, p.curUnsafeKey.Key, p.curRawKey, p.meta.RawBytes)
+		//
+		// TODO(ssd): Should we error if we find an inline when
+		// ReturnRawMVCCValues is set? Anyone scanning with that option
+		// set should not be encountering inline values.
```

Comment:
I don't think we've ever considered whether inline values can have an MVCCValueHeader. I think there's no reason to assume they couldn't, but it's probably a door we should close until we decide that we want to open it explicitly. _However_ if for whatever reasons inline puts with a header are out there, we want that to keep working. So my vote would be to see (perhaps by adding an assertion plus an audit) whether there is a way for inline writes to end up with a header. If not - great, shut the door to it by adding an error check on the write path and an non-fatal and perhaps testbuild-only assertion on the read path.
I wouldn't do this in this PR though. You could file an issue.

Reply:
> I think for that to happen in the code as it is right now, there would have to be some caller constructing an extended-encoded value, and then sending us that value. Otherwise, the underlying storage function evaluates blind puts without looking at the write options that produce MVCCValueHeaders.

Okay. Sounds like we want to error out under the test build tag here then if we see extended encoding (to avoid any overhead in prod) and add a similar (test-only) check somewhere down in the write path along with a unit test that hits the assertion? I'd be okay with an issue filed about this and referenced here and mentioned in clarifying comments on Put explaining that only the simple value encoding is allowed (as well as a comment on the ReturnRawValues option that it is not compatible with inline reads).